### PR TITLE
add a feature and fix a problem with dialogs and popups

### PR DESCRIPTION
**Pull request by freaksdd**
I wrote two issues before but then i handle it by my self, cause i found a solution for this problems.
### feature:
- now the spacebar get a translation. Maybe there is not the right place, but for now i think there is okay.

### Problem with Dialog and Popups:
**Problem:**
cuteKeyboard can not be use with Dialog and Popup. The Modal do the keyboard unusable cause of there excution.

**Solution:**
register the QQuickItem of InputPanel in the Backend and set the focusItem as parent of the InputPanel

### CODE REVIEW

File: yocto-recipes/cutekeyboard_git.bb [hunk 1]
```diff
@@ -7,6 +7,6 @@ LIC_FILES_CHKSUM = "file://LICENSE;md5=2f6f4d4f7d76b223f08e17122d04010f"
 DEPENDS += "qtbase qtdeclarative qtmultimedia qtquickcontrols qtsvg qtxmlpatterns"
 
 SRCREV = "${AUTOREV}"
-SRC_URI = "git://github.com/amarula/cutekeyboard.git;protocol=https;branch=main"
+SRC_URI = "git://github.com/freaksdd/cutekeyboard.git;protocol=https;branch=main"
```
Comment (AndreaRicchi) [inline yocto-recipes/cutekeyboard_git.bb:10]: Changing the recipe URI cannot be merged. Revert this one

**Review by AndreaRicchi — CHANGES_REQUESTED** [review]
Looks good, thanks for the contribution. Just revert the Yocto recipe

**Comment by freaksdd** [comment]
I revert the change in Yocto recipe. Sorry for it. My knowledgement for versioncontrol are basic. So i don't understand with merges or request's from fork.

**Comment by AndreaRicchi** [comment]
@freaksdd I've tested the change in the example with this code:

```
    MessageDialog {
        id: dialog

        contentItem: RowLayout {
            TextField {
                placeholderText: ".."
            }

            Button {
                text: "ok"
                onClicked: close()
            }

        }

    }
```

but it doesn't work. Did you test with this kind of dialog?

**Comment by AndreaRicchi** [comment]
Also, instead of having two commits

image

one that changes and one that reverts, remove both for a clearer history.

Thanks

**Comment by freaksdd** [comment]
Hi, first, i' m from germany and my english are not so very well.
I have only test it with Dialog{} and Popup{}.
MessageDialog currently untestet.
In therortic it have to be work, but i will look at the problem and tell
you whats happen.

Am Do., 29. Aug. 2024 um 08:32 Uhr schrieb Andrea Ricchi <
***@***.***>:

**Comment by freaksdd** [comment]
i'm a beginnger with git commands.
i have try:
git reset --hard 926f8fea54a2519047ec45258074e7ac0e6bab3c
   output:
     HEAD is now at 926f8fe Add spaceIdentifier to QwertyLayout
git push --force
   output:
     Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
remote:
remote: Create a pull request for 'master' on GitHub by visiting:
remote:      https://github.com/freaksdd/cutekeyboard/pull/new/master
remote:
To https://github.com/freaksdd/cutekeyboard
 * [new branch]      master -> master
branch 'master' set up to track 'origin/master'.

is this done?


Am Do., 29. Aug. 2024 um 08:34 Uhr schrieb Andrea Ricchi <
***@***.***>:

**Comment by freaksdd** [comment]
i found it out and this two commits are deleted.

Am Do., 29. Aug. 2024 um 11:09 Uhr schrieb Peter Gabriel <
***@***.***>:

**Comment by freaksdd** [comment]
Hi Andrea Ricchi,
now i was able to be test your Dialog. You are right. It is does not run
with this dialog.
The Dialog is block the hole window so i have to check whats happen.

Am Do., 29. Aug. 2024 um 08:32 Uhr schrieb Andrea Ricchi <
***@***.***>:

**Comment by freaksdd** [comment]
i looked at the api. i think the problem is, that the QtQuick.Dialogs is
Abstracted from QMessageBox an this will be instantiate a new window. My
(our) BackendClass can not handle new windows currently.
When you have an idea you are welcome. Currently i have not so much time to
handle this.

Am Do., 29. Aug. 2024 um 08:32 Uhr schrieb Andrea Ricchi <
***@***.***>:

**Review by AndreaRicchi — CHANGES_REQUESTED** [review]
Ok, still that is a good feature to be included. 

Can you just improve the git history, so squash the 72f46c897a503a774463a48e245babec8aff602d and d2f0b72f33058171350440d4a9e3da3189d4a420 and also the 214f261d497d1640b7f38d9ac030872f7e6848ba and 926f8fea54a2519047ec45258074e7ac0e6bab3c .

You can use `git rebase -i` command.

I think merging a new feature with different commits for the same change is incorrect.

**Comment by freaksdd** [comment]
Hi,

yes i could be do this, but not now. Give me some days and i will do this.
Dear Peter

Am Di., 17. Sept. 2024 um 08:40 Uhr schrieb Andrea Ricchi <
***@***.***>:

**Comment by AndreaRicchi** [comment]
Hi @freaksdd I've tried to fix the history but I don't want to merge it manually maybe losing your contribution score on the site, so I've squashed the commits with a new message:

```
commit e2cc2400855278c66f6bbf87f94fee71eab2e514 (HEAD -> freaksdd/main)
Author: freaksdd <peter.gabriel.pg@gmail.com>
Date:   Thu Aug 15 07:23:44 2024 +0200

    Add translations for the SpaceBar
    
    Added SpaceBar translation with 'spaceIdentifier' property.
    
    The translation can be added to the map inside
    DeclarativeInputEngine.cpp, with 'space' as fallback.

commit 67b24dc275c0ccb560d08c80d3cafb723c0e4081
Author: freaksdd <peter.gabriel.pg@gmail.com>
Date:   Thu Aug 15 07:35:09 2024 +0200

    fix: delete old item-connection
    
    when there are manny items, there are a lot of connections in backend.
    if the current item in action, the connection can be handle.
    if a new item is focus, the 'old' connection can be overwritten.

commit a11aafa4976d879df7fb9bcc30fa54799ffbded6
Author: freaksdd <peter.gabriel.pg@gmail.com>
Date:   Wed Aug 14 14:25:58 2024 +0200

    Make Cutekeyboard compatible with Dialogs and Popup
    
    Because of the Modal option, the keyboard doesn't work with Dialogs
    and Popups.
    
    The solution is to register the QQuickItem of InputPanel
    in the Backend and set the focusItem as parent of the InputPanel.
    
    Note that the keyboard is still incompatible with QtQuick.Dialogs
``` 

I've used `git rebase -i HEAD~5`, moved the fix commit above and used the `squash` on the first 2 and latest 2.

I'll wait for your updates to merge in the main branch.

Regards

**Comment by freaksdd** [comment]
Hi Andrea,

i looked again at the history and my Github Desktop shows me, that the
commit is alwas be reverted.
[image: grafik.png]
So when there is not done, i need some help to change it so how do you like.

Dear Peter

Am Mi., 21. Aug. 2024 um 09:33 Uhr schrieb Andrea Ricchi <
***@***.***>:

**Comment by freaksdd** [comment]
so i have think about my changes in the branch. It is possible to delete my
commits and i create a new branch to push it there?
I don't know what i done. I had think, the my fork doesn'tn matter with
your repository. So what do you think about?

Dear Peter

Am Fr., 20. Sept. 2024 um 15:37 Uhr schrieb Andrea Ricchi <
***@***.***>:

**Comment by AndreaRicchi** [comment]
@freaksdd I've manually updated the commit messages and merged them.

Thanks

**Comment by freaksdd** [comment]
Hi Andrea,

Okay many thanks. I very sorry about the conflicts and Trouble i have made.
What ist the next step i can do?
Should i Check the Problem with the Dialog (from the example you have
shared with me)?

Andrea Ricchi ***@***.***> schrieb am Di., 24. Sept. 2024,
09:48:

File: src/VirtualKeyboardInputContext.cpp [hunk 2]
```diff
@@ -108,9 +118,36 @@ void VirtualKeyboardInputContext::setFocusObject(QObject *object) {
 
     bool AcceptsInput = d->FocusItem->inputMethodQuery(Qt::ImEnabled).toBool();
     if (!AcceptsInput) {
+        hideInputPanel();
         return;
     }
 
+    // set the focusItem as parent of InputPanel
```
Comment (dpurgin) [inline src/VirtualKeyboardInputContext.cpp:125]: Hello @AndreaRicchi 

Please compare the changes in this file with the patch to qtvirtualkeyboard: https://codereview.qt-project.org/c/qt/qtvirtualkeyboard/+/292472/10 at `src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp`. Apparently, the relevant changes were copied as-is. Could it be seen as a GPL violation?

**Pull request closed without merging**